modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/construction/ConstructionServiceSuite.scala

```diff
@@ -44,4 +59,51 @@ object ConstructionServiceSuite extends MutableIOSuite with Checkers {
       expect.eql(Left(InvalidPublicKey), _)
     }
   }
+
+  test("Successfully deserialize a transaction from hex") { res =>
+    implicit val securityProvider = res._1
+    implicit val kryo = res._2
+
+    val cs = ConstructionService.make[IO]()
+
+    KeyPairGenerator.makeKeyPair[IO].flatMap { srcKey =>
+      KeyPairGenerator.makeKeyPair[IO].flatMap { dstKey =>
+        val srcAddress = srcKey.getPublic.toAddress
+        val dstAddress = dstKey.getPublic.toAddress
+
+        generateTransactions(srcAddress, srcKey, dstAddress, 1).flatMap(_.map { hashedTransaction =>
+          val signedTransaction = hashedTransaction.signed
+
+          KryoSerializer[F]
+            .serialize(signedTransaction)
+            .map { bytes =>
+              val hex = Hex.fromBytes(bytes)
+
+              cs.deserializeTransactionFromHex(hex)
+                .rethrowT
+                .map(
+                  expect.eql(hashedTransaction.hash, _)
+                )
+            }
+            .toOption
+            .getOrElse(IO.pure(expect(false)))
+        }.head)
+      }
+    }
+  }
+
+  test("Failed to deserialize transaction from hex") { res =>
```

Comment:
accordingly, `returns MalformedTransaction for invalid transaction hex`

Reply:
Changed.